src/assets/wise5/authoringTool/edit-branch/edit-branch.component.ts

```diff
@@ -0,0 +1,260 @@
+            }
+          } else if (name === 'choiceChosen') {
+            createBranchCriterion = 'choiceChosen';
+            if (params != null && params.choiceIds != null && params.choiceIds.length > 0) {
+              branch.choiceId = params.choiceIds[0];
+              choices.push(params.choiceIds[0]);
+            }
+
+            if (createBranchNodeId && createBranchComponentId) {
+              const choices = this.projectService.getChoices(
+                createBranchNodeId,
+                createBranchComponentId
+              );
+              if (choices != null) {
+                branch.choices = copy(choices);
+              }
+            }
+          }
+        }
+      }
+
+      const nodeIdsInBranch = this.projectService.getNodeIdsInBranch(this.targetId, transition.to);
+      for (const nodeId of nodeIdsInBranch) {
+        const item = branch.items[nodeId];
+        if (item != null) {
+          item.checked = true;
+          branch.checkedItemsInBranchPath.push(item);
+        }
+      }
+    }
+
+    if (createBranchCriterion == null) {
+      if (this.node.transitionLogic.howToChooseAmongAvailablePaths === 'workgroupId') {
+        createBranchCriterion = 'workgroupId';
+      } else if (this.node.transitionLogic.howToChooseAmongAvailablePaths === 'random') {
+        createBranchCriterion = 'random';
+      } else if (this.node.transitionLogic.howToChooseAmongAvailablePaths === 'tag') {
+        createBranchCriterion = 'tag';
+      }
+    }
+    this.setCriteria(createBranchCriterion);
+    if (createBranchNodeId != null) {
+      this.setNodeId(createBranchNodeId);
+    }
+    if (createBranchComponentId) {
+      this.setComponentId(createBranchComponentId);
+    }
+
+    // use set timeout to give the form time to populate the path form group controls
+    setTimeout(() => {
+      if (scores.length > 0) {
+        for (let i = 0; i < scores.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(scores[i].join(', '));
+        }
+      }
+      if (choices.length > 0) {
+        for (let i = 0; i < choices.length; i++) {
+          this.pathFormGroup.controls[`path-${i + 1}`].setValue(choices[i]);
+        }
+      }
+    });
+    this.changeDetector.detectChanges();
+  }
+
+  protected removeBranchButtonClicked(): void {
+    if (
+      confirm(
+        $localize`Are you sure you want to remove the branch?\n\nThe branch structure will be removed but the steps will not be deleted.`
+      )
+    ) {
+      this.removeBranch();
+    }
+  }
+
+  private removeBranch(): void {
+    for (let bp = 0; bp < this.branchPaths.length; bp++) {
+      const branchPath = this.branchPaths[bp];
+      this.removeBranchPath(branchPath);
+      bp--; // shift the counter back one because we have just removed a branch path
+    }
+
+    const nodeId = this.targetId; // branch point node
+    const nodeIdAfter = this.projectService.getNodeIdAfter(nodeId);
```

Comment:
Use this.node.id instead here and below?
```suggestion
    const nodeIdAfter = this.projectService.getNodeIdAfter(this.node.id);
```